### PR TITLE
added a skeletonization step in the skeleton resampling process

### PR DESCRIPTION
**Pull request by denisri**
+ optimizations (avoid loops using C++ functions)

### CODE REVIEW

File: deep_folding/brainvisa/utils/resample.py [hunk 1]
```diff
@@ -112,36 +123,137 @@ def resample(input_image: Union[str, aims.Volume],
...
     # Note also that background shall not be given in values
 
-    # Create one bucket by value (except background)
-    # FIXME: Create several buckets because I didn't understood how to add
-    # several bucket to a BucketMap
-    for i, v in enumerate(values):
+    if aims.version() >= (5, 2):
+        reorder = False
         toc = time()
+        if values != sorted(np.unique(vol_dt[vol_dt != background])):
```
Comment (PierreAuriau) [inline deep_folding/brainvisa/utils/resample.py:129]: To clarify the comparison between the two arrays, I think that np.any() is needed.
In the case where the two arrays have different lengths, the comparison will raise an error in recent numpy versions (for previous versions of numpy, it is only a warning)

**Comment by denisri** [comment]
does it work better using list comparison ?

**Comment by PierreAuriau** [comment]
> does it work better using list comparison ?

Yes ! Sorry, I misunderstood the real issue.
I think that "values" should be turned into a list to use the index method further in the code.

**Review by JoelChavas — APPROVED** [review]
Tested by Julien. OK

**Pull request merged**